aspnet-server-1/src/IO.Swagger/Controllers/DeadCrumbsApi.cs

```diff
@@ -0,0 +1,167 @@
+            List<Location> locations = locationCollection.Find((u) => u.UserRef == username).ToList();
+            if (locations.Count == 0)
+            {
+                return StatusCode(404, $"No locations found!");
+            }
+            locations.Sort((loc1, loc2) => loc2.Timestamp.Value.CompareTo(loc1.Timestamp.Value));
+            Location newestLoc = locations[0];
+
+            var result = new ObjectResult(newestLoc);
+            return result;
+        }
+
+        /// <summary>
+        /// return the user
+        /// </summary>
+        /// <param name="username">The username of the user.</param>
+        /// <response code="200"> Succesfull request! Returns a user represented as JSON</response>
+        /// <response code="404"> No user found called "username"</response>
+        [HttpGet]
+        [Route("/User/{username}")]
+        [ValidateModelState]
+        [SwaggerOperation("GetUser")]
+        [SwaggerResponse(statusCode: 200, type: typeof(User), description: "a user represented as json")]
+        public virtual IActionResult GetUser([FromRoute][Required]string username)
+        {
+            var userCollection = db.GetCollection<User>("user");
+            User user = userCollection.Find((u) => u.Username == username).FirstOrDefault();
+            
+            if (user == null)
+            {
+                return StatusCode(404, $"No user found called \"{username}\"!");
+            }
+            var result = new ObjectResult(user);
+            return result;
+        }
+
+        /// <summary>
+        /// returns all user as a list
+        /// </summary>
+        /// <response code="200">Succesfull request! Returns a JSON array of users</response>
+        /// <response code="404"> No users found</response>
+        [HttpGet]
+        [Route("/Users")]
+        [ValidateModelState]
+        [SwaggerOperation("GetUsers")]
+        [SwaggerResponse(statusCode: 200, type: typeof(List<User>), description: "a JSON array of users")]
+        public virtual IActionResult GetUsers()
+        {
+            var userCollection = db.GetCollection<User>("user");
+            List<User> users = userCollection.Find(_ => true).ToList();
+            if (users.Count==0)
+            {
+                return StatusCode(404, $"No users found!");
+            }
+            var result = new ObjectResult(users);
+            return result;
+        }
+
+        /// <summary>
+        /// adds a new location
+        /// </summary>
+        /// <param name="location">a JSON object of a location</param>
+        /// <response code="201">Succesfull request! Created new location</response>
+        /// <response code="400">Bad request! Location not added</response>
+        [HttpPost]
+        [Route("/Location")]
+        [ValidateModelState]
+        [SwaggerOperation("PostLocation")]
+        public virtual IActionResult PostLocation([FromBody]Location location)
```

Comment:
Ifølge Swagger kan den også returnere kode 200, men ikke her

Reply:
Should be fixed